Source/relay/TourGuide/Items of the Month/2023/Closed Circuit Pay Phone.ash

```diff
@@ -0,0 +1,87 @@
+RegisterTaskGenerationFunction("IOTMClosedCircuitPayPhoneGenerateTasks");
+void IOTMClosedCircuitPayPhoneGenerateTasks(ChecklistEntry [int] task_entries, ChecklistEntry [int] optional_task_entries, ChecklistEntry [int] future_task_entries)
+{
+    if (!lookupItem("closed-circuit pay phone").have())
+        return;
+
+    string [int] affinityDescription;
+    string [int] questDescription;
+    string url = "inv_use.php?pwd=" + my_hash() + "&which=3&whichitem=11169";
+
+    int shadowLodestones = available_amount($item[Rufus's shadow lodestone]);
+    if (shadowLodestones > 0) {
+        questDescription.listAppend(HTMLGenerateSpanFont("Have " + pluralise($item[Rufus's shadow lodestone]) + ".", "purple"));
+    }
+
+    int riftAdvsUntilNC = get_property_int("encountersUntilSRChoice");
+    questDescription.listAppend(HTMLGenerateSpanFont(riftAdvsUntilNC + " encounters until NC/boss.", "black"));
+
+    if (get_property("questRufus").contains_text("step1")) {
```

Comment:
i think what i would prefer is if we built this with a bit more of a stated logic, where the subentry make uses a title that is status dependent (where we have comments noting what the statuses mean), the priority is dependent on that status, and the quest description build is a bit more compressed. that way the logic that populates the task entry is firmly tied to the logic chain and isn't quite as distributed as it is here.

Reply:
Okay, took a swing at this. I made it so the tile only gets added in a single place and I think the logic is a lot clearer about what happens in the various cases. The quest description is still spread out over a few places but IMO the single place where tile gets added still makes it clearer. Lemme know if this lines up with what you were thinking.